benchmark/scripts/consolidate.ts

```diff
@@ -0,0 +1,337 @@
+          isColdStart: false,
+          function: {},
+        };
+        break;
+
+      case isoDateRegex.test(message):
+        // console.log(`    - extract API call data`);
+        const apiData = getJson(message);
+        consolidatedEntry.name = apiData.context.name;
+        consolidatedEntry.runtime = apiData.context.runtime;
+        consolidatedEntry.sdk = apiData.context.sdk;
+        consolidatedEntry.latency = apiData.latency;
+        break;
+
+      case message.startsWith("REPORT"):
+        // console.log(`    - extract function call data`);
+        consolidatedEntry.function.duration = parseNumber(
+          extract(durationRegex, message)
+        );
+        consolidatedEntry.function.maxMemory = parseNumber(
+          extract(maxMemoryRegex, message)
+        );
+        const initDuration = parseNumber(extract(initDurationRegex, message));
+        consolidatedEntry.isColdStart = !!initDuration;
+        consolidatedEntry.function.initDuration = initDuration;
+        result.log.push(consolidatedEntry);
+        break;
+
+      default:
+        count.logEntries += 1;
+        break;
+    }
+  }
+
+  console.log(`- Build datasets`);
+  const datasetPointers: string[] = [];
+  for (const entry of result.log) {
+    const label = entry.name.substring(0, entry.name.lastIndexOf("-"));
+    const targetDataset = entry.isColdStart ? "coldStarts" : "warmStarts";
+
+    if (entry.isColdStart) {
+      // initDuration
+      if (!result.datasets.coldStarts.initDuration[label]) {
+        datasetPointers.push(`datasets.${targetDataset}.initDuration.${label}`);
+        result.datasets.coldStarts.initDuration["title"] =
+          "Cold start: init duration";
+        result.datasets.coldStarts.initDuration[label] = {
+          label: label,
+          order: CONFIG.functions.reduce(
+            (prev, curr) =>
+              curr.functionName === label ? curr.chart.order : prev,
+            0
+          ),
+          data: [entry.function.initDuration],
+        };
+      } else {
+        result.datasets.coldStarts.initDuration[label].data.push(
+          entry.function.initDuration
+        );
+      }
+
+      // Total duration
+      if (!result.datasets.coldStarts.totalDuration[label]) {
+        datasetPointers.push(
+          `datasets.${targetDataset}.totalDuration.${label}`
+        );
+        result.datasets.coldStarts.totalDuration["title"] =
+          "Cold start: total duration";
+        result.datasets.coldStarts.totalDuration[label] = {
+          label: label,
+          order: CONFIG.functions.reduce(
+            (prev, curr) =>
+              curr.functionName === label ? curr.chart.order : prev,
+            0
+          ),
+          data: [entry.function.initDuration + entry.function.duration],
+        };
+      } else {
+        result.datasets.coldStarts.totalDuration[label].data.push(
+          entry.function.initDuration + entry.function.duration
+        );
+      }
+    }
```

Comment:
True. This is part of the refactoring I mentioned previously. I posted the results quickly however I will do some refactoring now to avoid technical debt accumulation.